### PR TITLE
Use secure_getenv() when building using Emscripten

### PR DESCRIPTION
**Pull request by ahmedyarub**

### CODE REVIEW

File: kubernetes/config/incluster_config.c [hunk 1]
```diff
@@ -47,7 +47,7 @@ static int setBasePathInCluster(char **pBasePath)
 
 #if defined(WIN32) || defined(_WIN32) || defined(__WIN32__) || defined(__NT__) || defined(__APPLE__)
     const char *service_host_env = getenv(SERVICE_HOST_ENV_NAME);
-#elif __linux
+#elif __linux || defined(__EMSCRIPTEN__)
```
Comment (ityuhui) [inline kubernetes/config/incluster_config.c:50]: Which platform do you test on ?  Windows ?

Comment (ahmedyarub) [reply kubernetes/config/incluster_config.c:50]: Tested on Ubuntu and Windows. Compiled using latest emsdk.

Comment (ityuhui) [reply kubernetes/config/incluster_config.c:50]: On Ubuntu,  it's enough to use `#elif __linux` to enable `secure_getenv`

On Windows, what's the precompiled result ？
```
#if defined(WIN32) || defined(_WIN32) || defined(__WIN32__) || defined(__NT__) || defined(__APPLE__)
    const char *service_host_env = getenv(SERVICE_HOST_ENV_NAME);
```
also takes effect.

Comment (ahmedyarub) [reply kubernetes/config/incluster_config.c:50]: I think that there is a misunderstanding here.
These flags refer to the target OS, not the host where compilation is taking place.
This fix is for targeting WASM32 when compiled using Emscripten. The resulting binary neither runs on Windows nor Linux. It runs inside the browser. __linux and WIN32 are not set when targeting Emscripten. 
https://en.wikipedia.org/wiki/WebAssembly

Comment (ityuhui) [reply kubernetes/config/incluster_config.c:50]: The #if preprocessor directive takes effect in the compilation process.

But I get it that Emscripten/WASM is neither Windows nor Linux.

The change is OK to me.

**Comment by ityuhui** [comment]
It seems EasyCLA blocks this PR.

**Comment by ahmedyarub** [comment]
> It seems EasyCLA blocks this PR. Could you handle it ? @ahmedyarub

I have already signed it. If you check the tags of this PR you will find that I've already signed it!

**Comment by ahmedyarub** [comment]
For some reason when I sign the CLA it redirects me to another PR!
https://github.com/kubernetes-client/c/pull/93

**Pull request merged**